test/routs/soldiersTest.js

```diff
@@ -0,0 +1,94 @@
+const mocha = require("mocha");
+const http = require("http");
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const expect = require('chai').expect;
+const configs = require('../../configFile');
+const fs = require("fs");
+
+let mongoServer;
+let client;
+let db;
+
+describe("soldiers api", function () {
+  const url = 'http://localhost:3000/soldiers';
+  let testData = 1;
+
+  before("soldiers api", function (done) {
+    require("../../start").serverUtils.then(res => {
```

Comment:
you are using a promise api here, which is fine. but you are also using a callback api which is also fine...but if you use both of them together I would have to kill you, pick one, your favourite and only use that one in any given methXMLHttpRequest od